### PR TITLE
fix every issue related to printf

### PR DESCRIPTION
**Pull request by nnyyxxxx**
# Warning: big diff incoming
# Read description before looking at the diff, it will explain why so many files are changing.
# Going to conflict with #573 
# Might conflict with #591 

## Type of Change
- [x] Refactoring

## Description
1. Increases synergy
2. Fixes problems with packager not being double quoted and because of that it unquotes itself (shellcheck noticed that)
3.  fixes issues with newlines being created on interactive prints
4. fixes an issue with the zsh script (shellcheck noticed that)
5. multiple quoting fixes
6. multiple formatting fixes

## Testing
tested

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] My changes generate no errors/warnings/merge conflicts.

## credits
- @adamperkowski for some fixes

### CODE REVIEW

File: core/tabs/applications-setup/Developer-tools/sublime-setup.sh [hunk 1]
```diff
@@ -27,9 +27,10 @@ installSublime() {
                     "$ESCALATION_TOOL" "$PACKAGER" install sublime-text
                     ;;
                 *)
-                    printf "%b\n" "${RED}The script does not support your Distro. Install manually..${RC}"
+                    printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
+                    exit 1
                     ;;
-            esac
+        esac
```
Comment (adamperkowski) [inline core/tabs/applications-setup/Developer-tools/sublime-setup.sh:33]:
```suggestion
            esac
```

File: core/tabs/applications-setup/browser-setup.sh [hunk 2]
```diff
@@ -227,7 +229,7 @@ browserSetup() {
     printf "%b\n" "7. Thorium"
     printf "%b\n" "8. Lynx"
     printf "%b\n" "----------------------------"
-    printf "%b\n"  "Enter your choices (e.g., 1 3 5): "
+    printf "%b"  "Enter your choices (e.g., 1 3 5): "
```
Comment (adamperkowski) [inline core/tabs/applications-setup/browser-setup.sh:232]:
```suggestion
    printf "%b"  "Enter your choices (e.g. 1 3 5): "
```

File: core/tabs/utils/auto-login.sh [hunk 3]
```diff
@@ -15,7 +15,7 @@ list_sessions() {
     printf "%b\n" "8) Openbox (openbox.desktop)"
     printf "%b\n" "9) i3 (i3.desktop)"
     printf "%b\n" "10) Custom session"
-    printf "%b\n" "Enter your choice [1-10]: "
+    printf "%b" "Enter your choice [1-10]: "
```
Comment (adamperkowski) [inline core/tabs/utils/auto-login.sh:18]:
```suggestion
    printf "%b" "Enter your choice (1-10): "
```

File: core/tabs/utils/auto-login.sh [hunk 4]
```diff
@@ -125,7 +125,7 @@ configure_or_remove_autologin() {
     printf "%b\n" "Do you want to add or remove autologin?"
     printf "%b\n" "1) Add autologin"
     printf "%b\n" "2) Remove autologin"
-    printf "%b\n" "Enter your choice [1-2]: "
+    printf "%b" "Enter your choice [1-2]: "
```
Comment (adamperkowski) [inline core/tabs/utils/auto-login.sh:128]:
```suggestion
    printf "%b" "Enter your choice (1-2): "
```

File: core/tabs/utils/auto-login.sh [hunk 5]
```diff
@@ -134,7 +134,7 @@ configure_or_remove_autologin() {
         printf "%b\n" "2) GDM"
         printf "%b\n" "3) SDDM"
         printf "%b\n" "4) LXDM"
-        printf "%b\n" "Enter your choice [1-4]: "
+        printf "%b" "Enter your choice [1-4]: "
```
Comment (adamperkowski) [inline core/tabs/utils/auto-login.sh:137]:
```suggestion
        printf "%b" "Enter your choice (1-4): "
```

File: core/tabs/utils/auto-login.sh [hunk 6]
```diff
@@ -150,7 +150,7 @@ configure_or_remove_autologin() {
...
```
Comment (adamperkowski) [inline core/tabs/utils/auto-login.sh:153]:
```suggestion
        printf "%b" "Enter your choice (1-4): "
```

File: core/tabs/utils/create-bootable-usb.sh [hunk 7]
```diff
@@ -144,7 +144,7 @@ write_iso(){
     list_devices
 
     # Prompt user for USB device
-    printf "Enter the USB device (e.g., /dev/sdX): "
+    printf "%b" "Enter the USB device (e.g., /dev/sdX): "
```
Comment (adamperkowski) [inline core/tabs/utils/create-bootable-usb.sh:147]:
```suggestion
    printf "%b" "Enter the USB device (e.g. /dev/sdX): "
```

File: core/tabs/utils/encrypt_decrypt_tool.sh [hunk 8]
```diff
@@ -20,9 +20,8 @@ if ! command_exists openssl; then
             "$ESCALATION_TOOL" "$PACKAGER" install openssl
             ;;
         *)
-            printf "%b\n" "${RED}Your Linux distribution is not supported by this script.${RC}"
-            printf "%b\n" "${YELLOW}You can try installing OpenSSL manually:${RC}"
-            printf "%b\n" "1. Refer to your distribution's documentation."
+            printf "%b\n" "${RED}Unsupported package manager: "$PACKAGER"${RC}"
```
Comment (adamperkowski) [inline core/tabs/utils/encrypt_decrypt_tool.sh:23]:
```suggestion
            printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
```

File: core/tabs/utils/numlock.sh [hunk 9]
```diff
@@ -48,7 +48,7 @@ numlockSetup() {
     create_service
   fi
 
-  printf "%b\n" "Do you want to enable Numlock on boot? (y/n): "
+  printf "%b\n" "Do you want to enable Numlock on boot? (Y/n): "
   read -r confirm
   if [ "$confirm" = "y" ] || [ "$confirm" = "Y" ]; then
     "$ESCALATION_TOOL" systemctl enable numlock.service --quiet
```
Comment (adamperkowski) [inline core/tabs/utils/numlock.sh:54]: y is not the default here. Shouldn't be `Y`.

```suggestion
  printf "%b\n" "Do you want to enable Numlock on boot? (y/N): "
  read -r confirm
  if [ "$confirm" = "y" ] || [ "$confirm" = "Y" ]; then
    "$ESCALATION_TOOL" systemctl enable numlock.service --quiet
```

File: core/tabs/utils/ollama.sh [hunk 10]
```diff
@@ -87,8 +87,8 @@ run_model() {
     printf "%b\n" "${YELLOW}Custom Models${RC}"
     custom_models=$(ollama list | grep 'custom-model-prefix') 
 
-    printf "%b\n" "${YELLOW}Please select a model to run:${RC}"
-    printf "%b\n" "${YELLOW}Enter the number corresponding to the model or enter the name of a custom model:${RC}"
+    printf "%b" "Please select a model to run: "
```
Comment (adamperkowski) [inline core/tabs/utils/ollama.sh:90]:
```suggestion
    printf "%b" "Select a model to run: "
```

File: core/tabs/utils/service-control.sh [hunk 11]
```diff
@@ -100,7 +100,7 @@ add_service() {
     printf "%b\n" "Service $SERVICE_NAME has been created and is ready to be started."
 
     # Optionally, enable and start the service
-    printf "Do you want to start and enable the service now? (y/n)\n"
+    printf "%b" "Do you want to start and enable the service now? (Y/n): "
     read -r START_ENABLE
 
     if [ "$START_ENABLE" = "y" ]; then
```
Comment (adamperkowski) [inline core/tabs/utils/service-control.sh:106]: Same here.

```suggestion
    printf "%b" "Do you want to start and enable the service now? (y/N): "
    read -r START_ENABLE

    if [ "$START_ENABLE" = "y" ] || [ "$START_ENABLE" = "Y" ]; then
```

File: core/tabs/utils/timeshift.sh [hunk 12]
```diff
@@ -75,11 +75,11 @@ create_snapshot() {
...
+    printf "%b" "${CYAN}Enter the snapshot name you want to restore: ${RC}"
     read -r SNAPSHOT
-    printf "%b\n" "${CYAN}Enter the target device (e.g., /dev/sda1): ${RC}"
+    printf "%b" "${CYAN}Enter the target device (e.g., /dev/sda1): ${RC}"
     read -r TARGET_DEVICE
-    printf "%b\n" "${CYAN}Do you want to skip GRUB reinstall? (yes/no): ${RC}"
+    printf "%b" "${CYAN}Do you want to skip GRUB reinstall? (Y/n): ${RC}"
     read -r SKIP_GRUB
 
     if [ "$SKIP_GRUB" = "yes" ]; then
```
Comment (adamperkowski) [inline core/tabs/utils/timeshift.sh:85]:
```suggestion
    printf "%b" "${CYAN}Do you want to skip GRUB reinstall? (y/N): ${RC}"
    read -r SKIP_GRUB

    if [ "$SKIP_GRUB" = "y" ] || [ "$SKIP_GRUB" = "Y" ]; then
```

File: core/tabs/utils/timeshift.sh [hunk 13]
```diff
@@ -117,7 +117,7 @@ delete_snapshot() {
 # Function to delete all snapshots
 delete_all_snapshots() {
     printf "%b\n" "${RED}WARNING: This will delete all snapshots!${RC}"
-    printf "%b\n" "${CYAN}Are you sure? (yes/no): ${RC}"
+    printf "%b" "${CYAN}Are you sure? (y/N): ${RC}"
     read -r CONFIRMATION
 
     if [ "$CONFIRMATION" = "yes" ]; then
```
Comment (adamperkowski) [inline core/tabs/utils/timeshift.sh:123]:
```suggestion
    printf "%b" "${CYAN}Are you sure? (y/N): ${RC}"
    read -r CONFIRMATION

    if [ "$CONFIRMATION" = "y" ] || [ "$CONFIRMATION" = "Y" ]; then
```

**Review by adamperkowski — CHANGES_REQUESTED** [review]

**Comment by adamperkowski** [comment]
Looked through every single changed line. Apart from what I've requested, great work. Thanks.

File: core/tabs/utils/numlock.sh (hunk 9 above)
Comment (nnyyxxxx) [reply core/tabs/utils/numlock.sh:54]: remove the \n aswell

Comment (adamperkowski) [reply core/tabs/utils/numlock.sh:54]: Done

**Comment by nnyyxxxx** [comment]
ready for merge

**Review by adamperkowski — APPROVED** [review]

**Comment by adamperkowski** [comment]
Every single line approved.

File: core/tabs/system-setup/global-theme.sh [hunk 14]
```diff
@@ -18,30 +18,30 @@ install_theme_tools() {
             "$ESCALATION_TOOL" "$PACKAGER" -S --needed --noconfirm qt6ct kvantum
             ;;
         *)
-            printf "%b\n" "${RED}Unsupported package manager. Please install qt6ct and kvantum manually.${RC}\n"
+            printf "%b\n" "${RED}Unsupported package manager: "$PACKAGER"${RC}"
```
Comment (ChrisTitusTech) [inline core/tabs/system-setup/global-theme.sh:21]: Missing Double quote for PACKAGER

Comment (nnyyxxxx) [reply core/tabs/system-setup/global-theme.sh:21]: thanks for spotting that

Comment (nnyyxxxx) [reply core/tabs/system-setup/global-theme.sh]: fixed

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**